enhancements/sig-architecture/81-addon-lifecycle/kustomize-rollout-examples/02-rollout-strategy/clustermanagement-addon.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ClusterManagementAddOn
+metadata:
+  name: cool-cluster-scoped-addon-configuration
+spec:
+  rolloutStrategy:
+    strategy: RollingUpdateWithCanary
+    rollingUpdateWithCanary:
+      # clusters in this placement will receive any update first.  The ManagedClusterAddOn must be Available
+      # and not Degraded and have the LastObservedGeneration indicating it saw our update.
+      # I'm not completely certain the LastObservedGeneration semantic covers, "I have applied this configuration",
+      # but that's the goal.
```

Comment:
current `lastObservedGeneration` is just to indicate the config generation, to achieve this, we may need to chance the config reference in the managedclusteraddon to represent the config status

Reply:
> current lastObservedGeneration is just to indicate the config generation, to achieve this, we may need to chance the config reference in the managedclusteraddon to represent the config status

Hash of spec perhaps.  I think that in order to build the higher level constructs we need to know when particular configuration level has been achieved so that we can judge, "was this successful" to have the canary the original PR desired.